provider/thp_fragment_tool.py

```diff
@@ -0,0 +1,49 @@
+"""
+thp_fragment tool.
+
+This module is meant to copy, build and execute the thp_fragment tool.
+"""
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+
+
+dst_dir = "/var/tmp"
+test_bin = "/var/tmp/thp_fragment"
+source_file = "thp_fragment.c"
+source_package = "thp_fragment.tar.gz"
+
+
+def copy_tool(test):
+    host_path = os.path.join(
+        data_dir.get_deps_dir("thp_defrag_tool"), source_package
+    )
+    copy_cmd = "cp -rf %s %s" % (host_path, dst_dir)
+    if process.system(copy_cmd, ignore_status=True, shell=True) != 0:
```

Comment:
`shutil.copy2`

Reply:
Code updated !

Reply:
Now that a binary is being copied the function returns a permission denied:
`PermissionError: [Errno 13] Permission denied: '/var/tmp/thp_fragment.c'`

Reply:
Do you have logs? I applied the #3987 and tried in python console
```python
>>> import shutil
>>> shutil.copy2("/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/deps/thp_defrag_tool/thp_fragment.c", "/var/tmp/")
'/var/tmp/thp_fragment.c'
```

Reply:
@PaulYuuu
```
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/generic/tests/trans_hugepage_defrag.py", line 66, in run
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|     thp_fragment_tool.copy_tool(test)
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/provider/thp_fragment_tool.py", line 25, in copy_tool
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|     shutil.copy2(host_path, dst_dir)
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|   File "/usr/lib64/python3.9/shutil.py", line 444, in copy2
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|     copyfile(src, dst, follow_symlinks=follow_symlinks)
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|   File "/usr/lib64/python3.9/shutil.py", line 266, in copyfile
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR|     with open(dst, 'wb') as fdst:
[stdlog] 2024-04-01 15:15:06,978 avocado.test stacktrace       L0049 ERROR| PermissionError: [Errno 13] Permission denied: '/var/tmp/thp_fragment.c'
```

Reply:
Also tried with `shutil.copy` and no luck

Reply:
Fixed now, thanks @PaulYuuu